project/Build.scala

```diff
@@ -1242,16 +1242,26 @@ object Build {
       libraryDependencies += ("org.scala-js" %%% "scalajs-dom" % "1.1.0").cross(CrossVersion.for3Use2_13)
     )
 
-  def generateDocumentation(targets: Seq[String], name: String, outDir: String, ref: String, params: Seq[String] = Nil) =
+  def generateDocumentation(targets: Seq[String], name: String, outDir: String, ref: String, params: Seq[String] = Nil, addBootclasspath: Boolean = false) =
     Def.taskDyn {
       val distLocation = (dist / pack).value
       val projectVersion = version.value
       IO.createDirectory(file(outDir))
       val stdLibVersion = stdlibVersion(Bootstrapped)
+      val scalaLib = findArtifactPath(externalCompilerClasspathTask.value, "scala-library")
+      val dottyLib = (`scala3-library` / Compile / classDirectory).value
       // TODO add versions etc.
       def srcManaged(v: String, s: String) = s"out/bootstrap/stdlib-bootstrapped/scala-$v/src_managed/main/$s-library-src"
       def scalaSrcLink(v: String, s: String) = s"-source-links:$s=github://scala/scala/v$v#src/library"
       def dottySrcLink(v: String, s: String) = s"-source-links:$s=github://lampepfl/dotty/$v#library/src"
+      def bootclasspath: Seq[String] = if(addBootclasspath) Seq(
+        "-bootclasspath",
```

Comment:
Why do you need a separate bootclasspath given these libraries will be on the classpath anyway?

Reply:
We need to somehow pass path to standard library to the snippet compiler, otherwise we get `MissingCoreLibraryException". I've seen that sbt in task `doc` automatically passes this path in `bootclasspath` arg so I added support for this arg in scaladoc. Then I realized that in our sbt project we don't pass explicit bootclasspath to `generateDocumentation` task and snippet compiler didn't work. Previously I wanted to just pass `System.getProperty("java.class.path") but it led to errors because we got sbt libs on classpath.

Reply:
OK, but you should be able to use the regular -classpath and not necessarily -bootclasspath (which isn't really needed anymore)

Reply:
But `-classpath` doesn't contain or at least it didn't contain stdlib entry when this was implemented and without stdlib entry I can't instantiate Context. Or maybe is there other way to obtain path to stdlib?

Reply:
Which -classpath? It looks like -classpath isn't passed at all to scaladoc here, so I was just suggesting using that instead of -bootclasspath to pass the stdlib

Reply:
Actually maybe the issue is in the scaladoc shell script? Unlike the scalac shell scrpt it doesn't pass the `-Dscala.usejavacp=true` flag so that the JVM classpath is available to the compiler: https://github.com/lampepfl/dotty/blob/83e17f110bfaa2fe876bacc90eca208f23930f98/dist/bin/scalac#L77-L78 vs https://github.com/lampepfl/dotty/blob/83e17f110bfaa2fe876bacc90eca208f23930f98/dist/bin/scaladoc#L129

Reply:
that's a good point, gonna check that